src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -1716,7 +1719,7 @@ namespace smt::noodler {
             // The following axioms are redundant in the sense of completeness, but in the nested replace calls 
             // they can relate the contains predicate from the general replace (and thence the SAT solver can help a lot).
             literal cnt = mk_literal(m_util_s.str.mk_contains(s, a));
-            add_axiom({~cnt, mk_eq(v, t,false)});
+            add_axiom({~cnt, mk_literal(m.mk_not(m.mk_eq(s, a))), mk_eq(v, t,false)});
```

Comment:
Why was `v` replaced with `mk_literal(m.mk_not(m.mk_eq(s, a)))`? It just looks like a bit of a random change. If it is correct, maybe just check if the comments before are still correct.

Reply:
I added a comment